cpp/src/arrow/record_batch.cc

```diff
@@ -253,6 +253,26 @@ bool RecordBatch::ApproxEquals(const RecordBatch& other) const {
   return true;
 }
 
+Result<std::shared_ptr<RecordBatch>> RecordBatch::SelectColumns(const std::vector<int>& indices) const{
+  int n = static_cast<int>(indices.size());
+
+  std::vector<std::shared_ptr<Field>> fields(n);
+  std::vector<std::shared_ptr<Array>> columns(n);
+  
+  for (int i = 0; i < n; i++) {
+    int pos = indices[i];
+    if (pos < 0 || pos > num_columns() - 1) {
+      return Status::Invalid("Invalid column index ", pos, " to select columns.");
+    }
+    fields[i] = schema()->field(pos);
+    columns[i] = column(pos);
+  }
+  
+  auto schema = std::make_shared<arrow::Schema>(std::move(fields));
```

Comment:
Here we probably want to preserve the metadata of the schema (see the `Table::SelectColumns` method)

Reply:
Oops, completely missed that - will update now, thanks for highlighting!